### ISSUE TITLE
Interaction design for Use Cases

### ISSUE THREAD

**Issue by fkleedorfer** [researchstudio-sat/webofneeds#1708]
I imagine this functionality, but the resulting interaction design may be different in its user flow:

When clicking '+ Post', we should show a tiled view of possible topics, such as 'Taxi', 'Transport', 'Real Estate', 'Meet People', 'Any Offer/Demand', and upon selection show the post view with specific text and widgets that allow specifying that

The next step may be the selection of the role of the user in that topic (transporter/sender/recipient), or if there is only one role, this step is skipped. 

Next, the create view for that topic/role combination should be shown. 

Note: this has to work in parallel to the quite recent 'search' functionality, possibly allowing to reach the same create view via two routes (search and post)

Bonus points: find a better term than 'topics' for this feature

**Comment by m0ru** [comment]
Different use-cases (e.g. calling a taxi) will require different setups/workflows for the create-post view (amongst others). A solution for this could be to first implement #1733 and then have a list of mandatory and optional input components and an agnostic create-post view that just renders the components from those lists, collects the data and calls the create-need-function.

**Comment by AndresJCremades** [comment]
I've created a first functional prototype for “Add More Details“ feature. It’s focused on mobile design, and now I’ll proceed to develop a prototype on Marvel. I have in mind a design for cases where there’re too many details to add, and then they need to be grouped into categories. Also, I have a couple of ideas for desktop views. It doesn’t differ that much from the mobile design, but it takes profit of a bigger screen.

[Add.Details.Proposal.Mobile.Design.pdf](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5b112afd4fe74600127128ff/125516c66e82c728ace21e0d46c9909e6da686bd8ee0b350ea6de29b39493472254a8e24728cef2df1667e12060810be5d4a1f06a3a3dc78e2a436628f5b0bbfd43c5c20c46e19fabfd04dbf01ba26894b83e4bb9850d2aa6c7bb3d4d216744b470a83ed5c3daa984eca5ffe4ee34453a0113362f43b.pdf)

What’s behind my proposal:

Staged Disclosure UI pattern

Goals:
	1.	Mobile Design First
	2.	Comfortable (Non-Overwhelming)
	3.	The user has to know where she/he is during the whole process of creating a search/post.

An optional step before publishing: an overview of the content to publish. This could be asked to the user: Do you want to see you post/search before publishing?

Sources: 
How to fix an overwhelming UI for reporting with too many options?
https://ux.stackexchange.com/questions/60351/how-to-fix-an-overwhelming-ui-for-reporting-with-too-many-options
NN/g - Progressive Disclosure (Staged Disclosure) 
https://www.nngroup.com/articles/progressive-disclosure/

**Comment by AndresJCremades** [comment]
I have developed two prototypes in Marvel App based on the first one created by @sigpie. One is the process of creation of a Post and the other the creation of a Search. Both have subtle differences from @sigpie 's original, who was going in the right direction in my opinion. 

Post: https://marvelapp.com/90a8gce
Search: https://marvelapp.com/90a8gce

**Comment by quasarchimaere** [comment]
@andres the urls are the same and show the post, also i am not that fond of the "in your face"-modal dialog for preview, i think it would be better to have a dedicated button in the create or search view instead (maybe without text and just an icon)

**Comment by AndresJCremades** [comment]
OK. URLs fixed.

Yeah, the dialogue window was a first random idea. I was thinking more of future bigger posts/searches, where the number of details could be larger, couldn't they? In those cases, I consider it recommendable having such feature. But yes, such windows can be annoying or tiresome after a few times, especially if takes no time creating the post/search. A dedicated button is a good idea. Something users notice and understand quickly. I'll come later with a proposal. Thanks!

**Comment by sigpie** [comment]
I agree with @quasarchimaere about the preview dialog. It's a good feature to have, but maybe not in this form. I'm also not sure about using full-area-colour to indicate details were selected. 

Using green by default is somewhat in opposition to having configurable colour schemes in the first place. 

Using the primary colour might send the wrong message, as we usually use the primary colour for interactable elements. But in this case, we want to signal that the interaction is already done. 

Using the secondary colour might work, but so far, we only use the secondary colour for the signup-button and not anywhere else. If we want to use the secondary colour to show that information has been entered, this probably apply to the whole user interface, not just the details picker.
